software/CONFIGURATION.md

```diff
@@ -1,55 +1,136 @@
   The hardware is capable of 12V maximum
-- `gate_voltage` is an integer in the range `[0, 12]` indicating the voltage that an output will produce when `cvx.on()` is called
+- `GATE_VOLTAGE` is an integer in the range `[0, 10]` indicating the voltage that an output will produce when `cvx.on()`
+  is called. This value must not be higher than `MAX_OUTPUT_VOLTAGE`
 
 
 
 # Experimental configuration
 
-Other configuration properties are used by [experimental features](software/firmware/experimental/__init__.py)
-and can be set using a similar static configuration file. This file is located at `/config/config_ExperimentalConfig.json` 
+Other configuration properties are used by [experimental features](/software/firmware/experimental/__init__.py)
+and can be set using a similar static configuration file. This file is located at `/config/ExperimentalConfig.json`
 on the Raspberry Pi Pico. If this file does not exist, default settings will be loaded.  The following
 shows the default configuration:
 
 ```json
 {
-    "volts_per_octave": 1.0,
+    "VOLTS_PER_OCTAVE": 1.0,
 }
 ```
 
-- `volts_per_octave` must be one of `1.0` (Eurorack standard) or `1.2` (Buchla standard)
+- `VOLTS_PER_OCTAVE` must be one of `1.0` (Eurorack standard) or `1.2` (Buchla standard)
+
+
+# Accessing config members in Python code
+
+The firmware converts the JSON file into a `ConfigSettings` object, where the JSON keys are converted
+to Python attributes.  The JSON object's keys must follow these rules, otherwise a `ValueError` will be raised:
+
+1. The string may not be empty
+1. The string may only contain letters, numbers, and the underscore (`_`) character
+1. The string may not begin with a number
+1. The string should be in `ALL_CAPS` -- this is not enforced, but is highly recommended
+
+The JSON key is converted into a Python attribute of the configuration object. For example, this JSON file
+```json
+{
+  "CLOCK_MULTIPLIER": 4,
+  "HARD_SYNC": true,
+  "WAVE_SHAPE": "sine"
+}
+```
+would produce a Python object with these attributes:
+```python
+>>> dir(config_object)
+[
+  '__class__',
+  '__init__',
+  '__module__',
+  '__qualname__',
+  '__dict__',
+  'to_attr_name',
+  'CLOCK_MULTIPLIER',
+  'HARD_SYNC',
+  'WAVE_SHAPE'
+]
+
+>>> config_object.CLOCK_MULTIPLIER
+4
+
+>>> config_object.HARD_SYNC
+True
+
+>>> config_object.WAVE_SHAPE
+'sine'
+```
+
+The `europi` namespace contains `.europi_config` and `.experimental_config` members that contain all of the
+configuration attributes described in the sections above:
+
+```python
+>>> from europi import europi_config
+>>> dir(europi_config)
+[
+  '__class__',
+  '__init__',
+  '__module__',
+  '__qualname__',
+  '__dict__',
+  'to_attr_name',
+  'CPU_FREQ',
+  'DISPLAY_CHANNEL',
+  'DISPLAY_HEIGHT',
+  'DISPLAY_SCL',
+  'DISPLAY_SDA',
+  'DISPLAY_WIDTH',
+  'EUROPI_MODEL',
+  'GATE_VOLTAGE',
+  'MAX_INPUT_VOLTAGE',
+  'MAX_OUTPUT_VOLTAGE',
+  'PICO_MODEL',
+  'ROTATE_DISPLAY'
+]
+```
+
+In code:
```

Comment:
In combination with the above comment, you could say here "For example, once you import the `europi_config` from the `europi` library, or import the whole library using `from europi import *`, you can read the configuration attributes like this"

Reply:
Done.